### ISSUE TITLE
Applying a forcedType prevents identity(true) from being generated

### ISSUE THREAD

**Issue by wcurrie** [jOOQ/jOOQ#10797]
### Expected behavior

After insert the primary key should be read back from mysql. Even for a primary key where the type is modified by a `forcedType` like
```xml
<forcedType>
  <name>BIGINT</name>
  <types>BIGINT UNSIGNED.*</types>
</forcedType>
```

### Actual behavior

In 3.14 the primary key is no longer read after calling `insert()`.

### Steps to reproduce the problem

- MCVE https://github.com/jOOQ/jOOQ-mcve/pull/6 (simpler) and https://github.com/jOOQ/jOOQ-mcve/pull/5 (with mysql) reproduce the problem. 

I've noticed two things.
1. The use of `forcedType` means `.identity(true)` is not output for the `TableField` of the generated `TableImpl`. This was already happening in 3.13.
2. In 3.14 it appears `getIdentity` stopped generating a reference to the table's key. Instead it outputs `super.getIdentity()`.

Perhaps propagating `identity=true` into a `forcedType` would make sense near [here](https://github.com/jOOQ/jOOQ/blob/9d5c87d35ac75d46f63bc511b7f44a5e52d8f54e/jOOQ-meta/src/main/java/org/jooq/meta/AbstractTypedElementDefinition.java#L307)? Assuming I've actually understood what is happening. My apologies if I have no idea what I'm talking about!

I acknowledge using `forcedType` on a primary key like in https://github.com/jOOQ/jOOQ-mcve/pull/5 is perhaps questionable but this issue showed up in an attempt to upgrade jooq at my employer.

### Versions

- jOOQ: 3.14
- Java: 11 but I don't think it will matter.
- Database (include vendor): Mysql
- OS: osx but I don't think it will matter.
- JDBC Driver (include name if inofficial driver): mysql connectorj

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce this.

As you've noticed, the issue has been there before, but it was not discovered, because the `DataType.identity()` information wasn't used for identity management logic. This will be fixed immediately. A 3.14.1 release should be out, soon.

**Comment by lukaseder** [comment]
I can confirm, that line (among others) that you've referenced, needs to be fixed. Thanks for the heads-up!

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0. Backport scheduled for 3.14.1 (#10800).

Thanks again for your report!
